plugins/traction_innkeeper/traction_innkeeper/v1_0/innkeeper/routes.py

```diff
@@ -522,6 +532,60 @@ async def tenant_create_token(request: web.BaseRequest):
     return web.json_response({"token": token})
 
 
+@docs(
+    tags=["multitenancy"],
+    summary="Get auth token for a subwallet (innkeeper plugin override)",
+)
+@request_schema(CreateWalletTokenRequestSchema)
+@response_schema(CreateWalletTokenResponseSchema(), 200, description="")
+@error_handler
+async def tenant_wallet_create_token(request: web.BaseRequest):
+    context: AdminRequestContext = request["context"]
+    wallet_id = request.match_info["wallet_id"]
+    wallet_key = None
+
+    mgr = context.inject(TenantManager)
+    profile = mgr.profile
+
+    # Tenants must always be fetch by their wallet id.
+    async with profile.session() as session:
+        rec = await TenantRecord.query_by_wallet_id(session, wallet_id)
+        LOGGER.warn("when creating token ", rec)
+        if rec.state == TenantRecord.STATE_DELETED:
+            raise web.HTTPUnauthorized(reason="Tenant is disabled")
+
+    # The rest is from https://github.com/hyperledger/aries-acapy-plugins/blob/main/multitenant_provider/multitenant_provider/v1_0/routes.py
+    LOGGER.warn(f"wallet_id = {wallet_id}")
+
+    # "builtin" wallet_create_token uses request.has_body / can_read_body
+    # which do not always return true, so wallet_key wasn't getting set or passed
+    # into create_auth_token.
+
+    # if there's no body or the wallet_key is not in the body,
+    # or wallet_key is blank, return an error
+    if not request.body_exists:
+        raise web.HTTPUnauthorized(reason="Missing wallet_key")
+
+    body = await request.json()
+    wallet_key = body.get("wallet_key")
+    LOGGER.warn(f"wallet_key = {wallet_key}")
```

Comment:
Would remove this entirely, it's `debug` in the Multitenant Provider, but logging the key in any situation, even debug, probably not good IMO (might want to remove that debug one in the other plugin)